sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1897,6 +1897,18 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  /**
+   * SPARK-38809 - Config option to allow skipping null values for hash based stream-stream joins.
+   * Its possible for us to see nulls if state was written with an older version of Spark,
+   * the state was corrupted on disk or if we had an issue with the state iterators.
+   */
+  val STATE_STORE_SKIP_NULLS_FOR_STREAM_STREAM_JOINS =
+  buildConf("spark.sql.streaming.stateStore.skipNullsForStreamStreamJoins.enabled")
+    .internal()
+    .doc("When true, this config will skip null values in hash based stream-stream joins.")
+    .booleanConf
```

Comment:
I missed one thing - let's tag version. Since this is a part of correctness issue fix and I plan to add this to 3.3 as well, please add `.version("3.3.0")`.

Reply:
cc. @MaxGekk I plan to get this into Spark 3.3, since it's a part of correctness bug fix.

Reply:
Done